custom_components/lifx_ceiling/light.py

```diff
@@ -93,7 +98,51 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on the downlight."""
         duration = int(kwargs.get(ATTR_TRANSITION, 0))
-        color = hsbk_for_turn_on(self._device.downlight_color, **kwargs)
+        # Use stored color as base instead of hardware state
+        stored_color = self.coordinator.get_downlight_color(self._device)
+
+        # Check if light is currently off (virtual state)
+        is_on = self.coordinator.get_downlight_is_on(self._device)
+
+        if not is_on:
+            # Light is off - check if this is just a brightness/color adjustment
+            # (no actual turn-on intent) vs an explicit turn-on request.
+            # is_adjustment_only is True when kwargs has values AND all keys are
+            # color/brightness related (no unknown keys that might indicate intent)
```

Comment:
I'm not sure I understand how/when calling `async_turn_on` isn't in itself a clear indication of intent. It would be super helpful if you could provide some additional clarity on this.